pkg/codegen/java/gen.go

```diff
@@ -1129,8 +1131,11 @@ func (mod *modContext) genResource(ctx *classFileContext, r *schema.Resource, ar
 			}
 		}
 		fprintf(w, "            .build();\n")
-		fprintf(w, "    }\n")
+	} else {
+		fprintf(w,
+			"        return args == null ? %s.Empty : args;\n", ctx.ref(argsFQN))
```

Comment:
Not really related to this fix, but `Empty` really shouldn't be used here. In my opinion we should use the builder, so to get 'required' property validation.

Reply:
I don't disagree but wonder if this would be a breaking change -- for sure it feels like the "right" thing to do but wonder if it would upset anything.